### PR TITLE
feat: adding api registry and component

### PR DESCRIPTION
**Pull request by CristhianF7**
- Adding API registry
- Adding API component using the API chart https://github.com/kubefirst/charts/tree/gh-pages/charts/api

### CODE REVIEW

File: components/api/application.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: api
+  namespace: argocd
```
Comment (converge) [inline components/api/application.yaml:5]: @jarededwards shall we move console related to console namespace?

Comment (jarededwards) [reply components/api/application.yaml:5]: I vote for namespace `kubefirst` - the console and api can share the same namespace

File: components/api/application.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                - path: /
+                  pathType: ImplementationSpecific
+          tls:
+            - secretName: api-tls
+              hosts:
+                - api.<AWS_HOSTED_ZONE_NAME>
+        
+  destination:
+    server: https://kubernetes.default.svc
+    namespace: api
```
Comment (jarededwards) [inline components/api/application.yaml:38]:
```suggestion
    namespace: kubefirst
```

File: components/api/application.yaml (hunk 1 above)
Comment (jarededwards) [inline components/api/application.yaml:5]:
```suggestion
  namespace: argocd
```

**Review by jarededwards — APPROVED** [review]

**Pull request merged**